src/zenml/materializers/default_materializer_registry.py

```diff
@@ -84,7 +100,7 @@ def get_materializer_types(
 
     def is_registered(self, key: Type[Any]) -> bool:
         """Returns if a materializer class is registered for the given type."""
-        return key in self.materializer_types
+        return any([issubclass(key, t) for t in self.materializer_types])
```

Comment:
`any()` will also accept a generator instead of a list, so you can simply remove the square brackets and it will still work (and be more efficient).

Reply:
Nice catch! Changed it accordingly.